src/deferred.js

```diff
@@ -182,6 +187,12 @@ jQuery.extend( {
 							if ( depth ) {
 								process();
 							} else {
+
+								// Call an optional hook to record the stack, in case of exception
+								// since it's otherwise lost when execution goes async
+								if ( jQuery.Deferred.getStackHook ) {
+									process.stackTrace = jQuery.Deferred.getStackHook();
```

Comment:
I struggled with finding a good place to put the stack trace, it needs to be per-promise and not leak memory.